docs/api/portable-stories.md

```diff
@@ -0,0 +1,184 @@
+
+### Return
+
+Type: `Record<string, ComposedStoryFn>`
+
+An object where the keys are the names of the stories and the values are the composed stories.
+
+Additionally, the composed story will have the following properties:
+
+| Property   | Type                                       | Description                                 |
+| ---------- | ------------------------------------------ | ------------------------------------------- |
+| storyName  | `string`                                   | The story's name                            |
+| args       | `Record<string, any>`                      | The story's args                            |
+| argTypes   | `ArgType`                                  | The story's argTypes                        |
+| id         | `string`                                   | The story's id                              |
+| parameters | `Record<string, any>`                      | The story's parameters                      |
+| load       | `() => Promise<void>`                      | Executes all the loaders for a given story  |
+| play       | `(context) => Promise<void>  \| undefined` | Executes the play function of a given story |
+
+## composeStory
+
+You can use `composeStory` if you wish to compose a single story for a component.
+
+```tsx
+// Button.test.ts|tsx
+import { render, screen } from '@testing-library/react';
+import { composeStory } from '@storybook/react';
+import { vi, test, expect } from 'vitest';
+import meta, { Primary as PrimaryStory } from './Button.stories';
+
+test('onclick handler is called', () => {
+  // Returns a component that already contain all decorators from story level, meta level and global level.
+  const Primary = composeStory(PrimaryStory, meta);
+
+  const onClickSpy = vi.fn();
+  render(<Primary onClick={onClickSpy} />);
+  const buttonElement = screen.getByRole('button');
+  buttonElement.click();
+  expect(onClickSpy).toHaveBeenCalled();
+});
+```
+
+### Type
+
+<!-- prettier-ignore-start -->
+```ts
+(
+  story: Story export,
+  componentAnnotations: Meta,
+  projectAnnotations?: ProjectAnnotations,
+  exportsName?: string
+) => ComposedStoryFn
+```
+<!-- prettier-ignore-end -->
+
+### Parameters
+
+#### `story`
+
+(**Required**)
+
+Type: `Story export`
+
+Specifies which story you want to compose.
+
+#### `componentAnnotations`
+
+(**Required**)
+
+Type: `Meta`
+
+The default export from the stories file containing the [`story`](#story).
+
+#### `projectAnnotations`
+
+Type: `ProjectAnnotation | ProjectAnnotation[]`
+
+This parameter is provided for convenience. You should likely use [`setProjectAnnotations`](#setprojectannotations) instead. Details about the `ProjectAnnotation` type can be found in that function's [`projectAnnotations`](#projectannotations-2) parameter.
+
+#### `exportsName`
+
+Type: `string`
+
+TK - What is this?
```

Comment:
During the story composition, the `exportsName` parameter will be used to create a property called `storyName`. This is useful for snapshot testing and the like, because users can refer to the `storyName` property to distinguish their snapshots. Here's how the name is determined internally:

```ts
const storyName =
    exportsName ||
    storyAnnotations.storyName ||
    storyAnnotations.story?.name ||
    storyAnnotations.name ||
    'unknown';
```

The `composeStory` function is used internally by `composeStories`, and `composeStories` has access to the exports name of the named exports and therefore _passes them down automatically_ to `composeStory`. In this case, users don't have to think about this.

However, when **not** using `composeStories`, but rather `composeStory`, the user is passing just the reference of a story. In this case, they might need to pass a custom exportName property if they do something that needs a specific story name or id. So the exportsName parameter is there in case they want to override it to something else. There was a single use case in a Github issue where someone from MS wanted this badly, so it was implemented: https://github.com/storybookjs/storybook/pull/22471

In general I think people don't even need to know this exists!

Reply:
Cases like this are tricky, where the nuance and details require a lengthy explanation as you've provided (thank you!) while that same lengthy explanation would give it far too much prominence in our docs. Ultimately, I opted to keep it as brief as possible. Do you think this is comprehensive enough?

> You probably don't need this. Because `composeStory` accepts a single story, it does not have access to the name of that story's export in the file (like `composeStories` does). If you must ensure unique story names in your tests and you cannot use `composeStories`, you can pass the name of the story's export here.

Reply:
Yup that sums it up pretty nicely!